misc/python/materialize/scalability/operations_test.py

```diff
@@ -0,0 +1,49 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+
+import time
+
+from psycopg import Cursor
+
+from materialize.scalability.operation import Operation
+
+
+class NoQuery(Operation):
+    def execute(self, cursor: Cursor) -> None:
+        pass
+
+
+class EmptyQuery(Operation):
+    def query(self) -> str:
+        return ""
+
+
+class SleepInPython(Operation):
+    def __init__(self, duration: float) -> None:
+        self.duration = duration
+
+    def execute(self, cursor: Cursor) -> None:
+        time.sleep(self.duration)
+
+
+class SleepInEnvironmentd(Operation):
+    def __init__(self, duration: float) -> None:
+        self.duration = duration
+
+    def query(self) -> str:
+        return f"SELECT mz_internal.mz_sleep({self.duration});"
+
+
+class SleepInClusterd(Operation):
+    def __init__(self, duration: float) -> None:
+        self.duration = duration
+
+    def query(self) -> str:
+        return f"SELECT mz_internal.mz_sleep(f1 * {self.duration}) FROM ((SELECT * FROM t1 WHERE f1 < 0) UNION ALL (SELECT 1));"
```

Comment:
Will this do the wait for every row in the data of `FROM`?

Reply:
The FROM is a UNION, the first part of which matches no rows. So it will be evaluated only once , for the `SELECT 1` part of the union. I am going to put comments about all that.

Reply:
I have added a comment.